runtime.go

```diff
@@ -271,7 +299,16 @@ func (r *runtime) Close(ctx context.Context) error {
 }
 
 // CloseWithExitCode implements Runtime.CloseWithExitCode
+//
+// Note: it also marks the internal `closed` field
 func (r *runtime) CloseWithExitCode(ctx context.Context, exitCode uint32) error {
+	r.mux.Lock()
```

Comment:
I don't think this lock is doing anything so we should remove it. the atomic compareAndSwap below will prevent multiple goroutines from passing.